File: MeetingScheduler/python/timeslot.py
from datetime import datetime
class TimeSlot:
  def __init__(self, start: datetime, end: datetime, id: str = None):
    self._start = start
    self._end = end
    self._id = id
  def get_start(self):
    return self._start
  def get_end(self):
    return self._end
  def get_id(self):
    return self._id
  def is_overlap(self, other: 'TimeSlot'):
    return self._start < other._end and self._end > other._start

File: MeetingScheduler/java/TimeSlot.java
package MeetingScheduler.java;

import java.time.LocalDateTime;

public class TimeSlot {
    private LocalDateTime start;
    private LocalDateTime end;
    private String id;

    public TimeSlot(LocalDateTime start, LocalDateTime end, String id) {
        this.start = start;
        this.end = end;
        this.id = id;
    }

    public String getId() {
        return this.id;
    }

    public LocalDateTime getStart() {
        return this.start;
    }

    public LocalDateTime getEnd() {
        return this.end;
    }

    public boolean isOverLap(TimeSlot other) {
        return this.start.isBefore(other.getEnd())
                && this.end.isAfter(other.getStart());
    }
}
